docs/schema_plugins.md

```diff
@@ -1 +1,234 @@
 # Extending NOMAD-Simulations to support custom methods and outputs
+
+As you develop your parser, you may find that the `nomad-simulations` package does not include some relevant quantities for your particular use case. You can easily extend upon the schema by adding your own custom schema under the `schema_packages/` directory in your parser plugin. For this, you should utilize and build upon `nomad-simulations` for consistency and future compatibility or integration. The following page demonstrates the basics of how to implement your own schema in practice.
+
+<div class="click-zoom">
+    <label>
+        <input type="checkbox">
+        <img src="../assets/schema_plugins/NOMAD_Simulations-extending-the-schema.png" alt="Schema extension decision tree." width="80%" title="Click to zoom in">
+    </label>
+</div>
+
+Within your parser plugin project, under `schema_packages/`, create a python file for your schema, e.g., `<parser_name>_schema.py`.
+
+Add the following imports to this file:
+
+```python
+import nomad_simulations
+import numpy as np
+from nomad.datamodel.data import ArchiveSection
+from nomad.metainfo import Context, MEnum, Quantity, Section, SectionProxy, SubSection
+```
+
+- `nomad_simulations`:
+
+- `numpy`:
+
+- `ArchiveSection`:
+
+- `SubSection`:
+
+-
+
+For the following examples we will use the H5MD-NOMAD file format, which is essentially a schema for molecular dynamics simulations, specified within the hdf5 file format. Imagine we have developed a parser for this file.
+
+Because the H5MD-NOMAD file is not created directly from a simulation program, there is additional metadata (beyond the program information), specifying the authorship of the file and the specifications for the generation (creator) of the H5MD schema used:
+
+    h5md
+     +-- version: Integer[2]
+     \-- author
+     |    +-- name: String[]
+     |    +-- (email: String[])
+     \-- creator
+     |    +-- name: String[]
+     |    +-- version: String[]
+     \-- program
+          +-- name: String[]
+          +-- version: String[]
+
+The program information can be stored under the existing `Program()` class within `nomad-simulations`. We already learned how to implement this into our parser:
+
+```python
+from nomad_simulations.schema_packages.general import Simulation, Program
+
+simulation = Simulation()
+simulation.program = Program(
+    name=group_h5md_dict.get('program_name'),
+    version=group_h5md_dict.get('program_version'),
+)
+```
+
+ However, we need to create a new class for the rest of the metadata. The "creator" information is referring to a secondary software used to create the hdf5 file, so it makes sense to store this in a class very similar to `Program()`. Let's first take a look at the current `Program()` schema:
```

Comment:
> The "creator" information is referring to a secondary software used to create the hdf5 file

This clarifies a lot, as "creator" sounds more like a person. I'd use the term "HDF5-generator" throughout to underscore this point. Else the semantics in your example feel off.